### DISCUSSION TITLE
Allow override of default text in headers, etc

### DISCUSSION THREAD

**Discussion by gullevek** [jfcherng/php-diff#66] [Ideas]
Currently there is a fixed Old/New in the Side by Side.

it would be nice to define that via an option setting, eg

```php
'header_text' => [
    'old' => '...',
    'new' => '...',
    'differences' => '...'
]
```

This would also free up language resource and move that to the user side where you could set whatever you want.

**Comment by jfcherng** [comment]
https://github.com/jfcherng/php-diff/blob/5eb2dabf4d44dd126f030ad17ccbc740921e7437/example/demo_base.php#L31-L33

> or an array which has the same keys with a language file

This is basically what you are trying to do.

```php
    'language' => [
	    'old' => '...',
	    'new' => '...',
	    'differences' => '...',
	],
```

**Reply by gullevek** [reply]
Yes, this would also work, but this would be a more breaking change as you now would need to check if it is a string (lang) or an array. So I thought adding a second override array might be easier

**Reply by jfcherng** [reply]
>  but this would be a more breaking change as you now would need to check if it is a string (lang) or an array. So I thought adding a second override array might be easier

This is not a breaking change and it's an existing feature.

---

I do see an issue that the user has to define all keys in that case. But I more prefer to make it into

```php
'language' => [
    // base
    'eng',
    // override
    [
        'old' => '...',
        'new' => '...',
        'differences' => '...',
    ],
	// override override etc...
],
```

**Reply by sloan58** [reply]
This might be documented elsewhere but I landed here while looking for this solution and found that the correct syntax is:

```php
'language' => [
    'eng',
    [
        'old_version' => '...',
        'new_version' => '...',
        'differences' => '...',
     ]
]
```

**Comment by gullevek** [comment]
Oh this makes so much more sense ... never thought about this.

**Comment by jfcherng** [comment]
[6.13.0](https://github.com/jfcherng/php-diff/releases/tag/6.13.0) has been released, which implements https://github.com/jfcherng/php-diff/discussions/66#discussioncomment-4712473

**Comment by gullevek** [comment]
Perfect! Thank you very much
